app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -386,10 +386,13 @@ def remove_middleware_datasource(ems_ref)
     end
 
     def emit_middleware_notification(type, op_name, op_arg, ems_ref)
+      mw_deployment = MiddlewareDeployment.find_by_ems_ref(ems_ref)
+      mw_server = mw_deployment.nil? ? MiddlewareServer.find_by_ems_ref(ems_ref) :
```

Comment:
Okay if we need to do two finds lets switch the `find_by_ems_ref` to `find_by(:ems_ref =>)`, and the ternary operator to an `if/else` per [rubocop](https://github.com/ManageIQ/manageiq/pull/13207#issuecomment-267399654) 

I don't know about middleware but most providers need `ems_id + ems_ref` to guarantee uniquness, do you need `find_by(:ems_id => ems_id, :ems_ref => ems_ref)` to find the server/deployment correctly?